src/coreclr/nativeaot/Runtime/Full/CMakeLists.txt

```diff
@@ -63,6 +66,24 @@ add_custom_command(
 
 set_source_files_properties("${CMAKE_CURRENT_BINARY_DIR}/AsmOffsets.inc" PROPERTIES GENERATED TRUE)
 
+set_source_files_properties(${SHARED_EVENTPIPE_SOURCES} PROPERTIES LANGUAGE CXX)
+set_source_files_properties(${SHARED_DIAGNOSTIC_SERVER_SOURCES} PROPERTIES LANGUAGE CXX)
+
+if(CLR_CMAKE_HOST_UNIX)
+  if (CMAKE_VERSION VERSION_GREATER 3.11 OR CMAKE_VERSION VERSION_EQUAL 3.11)
+    set_source_files_properties(${SHARED_EVENTPIPE_SOURCES} PROPERTIES COMPILE_OPTIONS -xc++)
+    set_source_files_properties(${SHARED_DIAGNOSTIC_SERVER_SOURCES} PROPERTIES COMPILE_OPTIONS -xc++)
+    add_definitions(-DHAVE_INTTYPES_H)
+  endif()
+endif(CLR_CMAKE_HOST_UNIX)
+
+add_definitions(-DFEATURE_PERFTRACING)
```

Comment:
Can you group this together with the rest of the `-DFEATURE` in the CmakeLists.txt one level up from here?